File: LittleDemo/DataBindingDemo/app/src/main/java/com/example/databindingdemo/MainActivity.java
package com.example.databindingdemo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.example.databindingdemo.databinding.ActivityMainBinding;
import com.example.databindingdemo.model.User;

public class MainActivity extends AppCompatActivity {

    private User user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this,R.layout.activity_main);
        user = new User("张三","123456");
        binding.setUserInfo(user);
        binding.tvUserName.setText("法外狂徒");
    }
}

File: LittleDemo/JetpackDemo/livedata/src/main/java/com/example/livedata/MainActivity.kt
package com.example.livedata

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.get
import com.example.livedata.databinding.ActivityMainBinding
import com.example.livedata.viewmodel.MyViewModel
import java.util.Timer
import java.util.TimerTask

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var viewModel: MyViewModel


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        viewModel = ViewModelProvider(
            this,
            ViewModelProvider.AndroidViewModelFactory(application)
        ).get()

        viewModel.getCurrentSecond()?.observe(
            this

        ){
            binding.textView.text = viewModel.getCurrentSecond()?.value.toString()

        }

        startTime()
    }

    private fun startTime() {
        Timer().schedule(object : TimerTask(){
            override fun run() {
                //异步线程用postValue()
                //UI线程用setValue()
                viewModel.getCurrentSecond()?.postValue(viewModel.getCurrentSecond()?.value?.plus(1))
            }

        },1000,1000)

    }
}